src/coprocessor/dag/dag.rs

```diff
@@ -48,7 +48,19 @@ impl<S: Snapshot + 'static> DAGContext<S> {
         req_ctx: ReqContext,
         batch_row_limit: usize,
     ) -> Result<Self> {
-        let mut eval_cfg = box_try!(EvalConfig::new(req.get_time_zone_offset(), req.get_flags(),));
+        let mut eval_cfg = box_try!(EvalConfig::new(req.get_flags()));
+
+        // We respect time zone name first, then offset.
+        if req.has_time_zone_name() {
+            box_try!(eval_cfg.set_time_zone_by_name(req.get_time_zone_name()));
+        } else if req.has_time_zone_offset() {
+            box_try!(eval_cfg.set_time_zone_by_offset(req.get_time_zone_offset()));
+        } else {
+            // This should be not reachable by design. However we still provide a default
+            // value, which is local.
+            box_try!(eval_cfg.set_time_zone_by_name("SYSTEM"));
```

Comment:
Seems we use `zone_offset = 0` as default before, I think we'd better keep the previous operation?

Reply:
Ok